packages/kit/src/runtime/telemetry/record_span.js

```diff
@@ -0,0 +1,71 @@
+/** @import { Attributes, Span, Tracer } from '@opentelemetry/api' */
+import { HttpError, Redirect } from '../control.js';
+import { load_status_code } from './load_otel.js';
+import { noop_span } from './noop.js';
+
+/**
+ * @template T
+ * @param {Object} options
+ * @param {string} options.name
+ * @param {Tracer} options.tracer
+ * @param {Attributes} options.attributes
+ * @param {function(Span): Promise<T>} options.fn
+ * @returns {Promise<T>}
+ */
+export async function record_span({ name, tracer, attributes, fn }) {
+	const SpanStatusCode = await load_status_code();
```

Comment:
Since `load_status_code` always dynamically imports `@opentelemetry/api`, this might be a performance issue. I'm not 100% sure though because I don't know if dynamic imports are cached. If they are, no problem, if not, you could manually cache the imported module instead instead.